smartsnippets/templates/smartsnippets/json_snippet_render.html

```diff
@@ -0,0 +1,226 @@
+{% extends "admin/base_site.html" %}
+{% load i18n admin_modify %}
+{% block stylesheet %}{% endblock %}
+
+{% block extrahead %}
+  <script type="text/javascript">
+   var current_site = {{current_site}};
+  </script>
+  {{ block.super }}
+  {{ media }}
+
+  <script type="text/javascript">
+   //<![CDATA[
+   (function($) {
+     function runOnLoad(f) {
+       if (runOnLoad.loaded){
+	 f();
+       }else{
+	 runOnLoad.funcs.push(f);
+       }
+     }
+     runOnLoad.funcs = [];
+     runOnLoad.loaded = false;
+     runOnLoad.run = function() {
+       if (runOnLoad.loaded) return;
+
+       for(var i = 0; i < runOnLoad.funcs.length; i++) {
+	 try {
+	   runOnLoad.funcs[i]();
+	 }catch(e) { }
+       }
+       runOnLoad.loaded = true;
+       delete runOnLoad.funcs;
+       delete runOnLoad.run;
+     };
+     if (window.addEventListener){
+       window.addEventListener("load", runOnLoad.run, false);
+     }else if (window.attachEvent){
+       window.attachEvent("onload", runOnLoad.run);
+     }else{
+       window.onload = runOnLoad.run;
+     }
+     runOnLoad(function(){
+       parent.setiframeheight($('body').height()+20, {{ plugin.pk }});
+     });
+
+     $(document).ready(function(){
+       $("a.add-another,a.related-lookup").each(function(i, sign){
+	 var href = $(sign).attr("href")
+	   if (href.substr(0,2) == ".."){
+	     href = "../../"+href
+	     $(sign).attr("href", href);
+	   }
+       });
+
+       var submitBtnName;
+
+       function is_json_field(element) {
+         var json_field_classes = ["merlinfield_hidden"],
+             classes = element.attr('class').split(/\s+/),
+             index;
+         for (index=0; index<json_field_classes.length; index++) {
+           if (classes.indexOf(json_field_classes[index]) !== -1) {
+             return true;
+           }
+         }
+         return false;
+       }
+
+       SnippetWidgetRegistry.initializeVariables();
+       $("#{{ opts.model_name }}_form").find(':submit').click(function(){
```

Comment:
Yes, this is a better option. We should make this change gradually.
Most of the JS code is copied from the old smart snippet admin rendering template and the indent is that we will not use this version of the smart snippets long term, but this is still a change we should make for the first version of Bento 3.